src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -187,6 +196,28 @@ void deleteUser(String uid) throws FirebaseAuthException {
     }
   }
 
+  DownloadAccountResponse listUsers(int maxResults, String pageToken) throws FirebaseAuthException {
+    ImmutableMap.Builder<String, Object> builder = ImmutableMap.<String, Object>builder()
+        .put("maxResults", maxResults);
+    if (pageToken != null) {
+      checkArgument(!pageToken.isEmpty(), "invalid end of list page token");
+      builder.put("nextPageToken", pageToken);
+    }
+
+    DownloadAccountResponse response;
+    try {
+      response = post("downloadAccount", builder.build(), DownloadAccountResponse.class);
+    } catch (IOException e) {
+      throw new FirebaseAuthException(LIST_USERS_ERROR,
+          "IO error while downloading user accounts.", e);
+    }
+    if (response == null) {
```

Comment:
Nit: this might be more readable if you move this into the try block.

Reply:
Done